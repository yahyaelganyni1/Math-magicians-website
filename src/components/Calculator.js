import { useState } from 'react';
import calculate from './logic/calculate';

export default function Calculator() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleChange = () => {};

  const handleClick = (e) => {
    setState({ ...state, ...calculate(state, e.target.outerText) });
  };
  const { next, total } = state;
  return (
    <div>
      <div className="calc-header">
        <h3>lets do some math</h3>
      </div>
      <div className="Calculator">
        <input
          type="text"
          value={next || total || '0'}
          className="answer"
          disabled="disabled"
          onChange={handleChange}
          style={{ direction: 'rtl' }}
        />
        <button type="button" onClick={handleClick}>
          AC
        </button>
        <button type="button" onClick={handleClick}>
          +/−
        </button>
        <button type="button" onClick={handleClick}>
          %
        </button>
        <button type="button" onClick={handleClick} className="sacndry-color">
          ÷
        </button>
        <button type="button" onClick={handleClick}>
          8
        </button>
        <button type="button" onClick={handleClick}>
          7
        </button>
        <button type="button" onClick={handleClick}>
          9
        </button>
        <button type="button" onClick={handleClick} className="sacndry-color">
          x
        </button>
        <button type="button" onClick={handleClick}>
          4
        </button>
        <button type="button" onClick={handleClick}>
          5
        </button>
        <button type="button" onClick={handleClick}>
          6
        </button>
        <button type="button" onClick={handleClick} className="sacndry-color">
          -
        </button>
        <button type="button" onClick={handleClick}>
          1
        </button>
        <button type="button" onClick={handleClick}>
          2
        </button>
        <button type="button" onClick={handleClick}>
          3
        </button>
        <button type="button" onClick={handleClick} className="sacndry-color">
          +
        </button>
        <button type="button" onClick={handleClick} className="zero">
          0
        </button>
        <button type="button" onClick={handleClick}>
          .
        </button>
        <button type="button" onClick={handleClick} className="sacndry-color">
          =
        </button>
      </div>
    </div>
  );
}
