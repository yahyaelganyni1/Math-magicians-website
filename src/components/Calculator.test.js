import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('expect Calculator to match snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
