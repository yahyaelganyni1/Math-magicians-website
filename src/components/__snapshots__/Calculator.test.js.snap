// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect Calculator to match snapshot 1`] = `
<div>
  <div
    className="calc-header"
  >
    <h3>
      lets do some math
    </h3>
  </div>
  <div
    className="Calculator"
  >
    <input
      className="answer"
      disabled="disabled"
      onChange={[Function]}
      style={
        Object {
          "direction": "rtl",
        }
      }
      type="text"
      value="0"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/−
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="sacndry-color"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="sacndry-color"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="sacndry-color"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="sacndry-color"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="sacndry-color"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
